frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx

```diff
@@ -9,44 +11,82 @@ import {
   AlertActionCloseButton,
 } from '@patternfly/react-core';
 import { k8sCreate } from '@console/internal/module/k8s';
-import { WizardCommonProps } from './reducer';
+import { WizardCommonProps, WizardState } from './reducer';
 import { createSSPayload } from './payloads';
-import { BackingStorageType, StepsId, RHCS } from '../../constants/create-storage-system';
-import { StorageSystemModel } from '../../models';
+import { BackingStorageType, Steps, StepsName } from '../../constants/create-storage-system';
+import { OCSServiceModel } from '../../models';
 import './create-storage-system.scss';
-import { ODF_EXTERNAL_PROVIDERS } from '../../odf-external-providers/external-providers';
+import {
+  getExternalStorage,
+  getStorageSystemKind,
+  getStorageSystemName,
+} from '../../utils/create-storage-system';
 
-const validateBackingStorageStep = (bsType, sc, externalProvider) => {
-  switch (bsType) {
+const validateBackingStorageStep = (backingStorage, sc) => {
+  const { type, externalStorage } = backingStorage;
+  switch (type) {
     case BackingStorageType.EXISTING:
       return !!sc;
     case BackingStorageType.EXTERNAL:
-      return !!externalProvider;
+      return !!externalStorage;
     case BackingStorageType.LOCAL_DEVICES:
       return true;
     default:
       return false;
   }
 };
 
-const validateStep = (id, state) => {
-  const { storageClass, backingStorage } = state;
-  const { type, externalProvider } = backingStorage;
-  switch (id) {
-    case StepsId.BackingStorage:
-      return validateBackingStorageStep(type, storageClass, externalProvider);
-    case StepsId.CreateLocalVolumeSet:
-    case StepsId.CapacityAndNodes:
-    case StepsId.ConnectionDetails:
-    case StepsId.CreateStorageClass:
-    case StepsId.ReviewAndCreate:
-    case StepsId.SecurityAndNetwork:
+const validateStep = (name: string, state: WizardState, t: TFunction) => {
+  const { storageClass, backingStorage, createStorageClass } = state;
+  const { externalStorage } = backingStorage;
+  const { canGoToNextStep } = getExternalStorage(externalStorage) || {};
+
+  switch (name) {
+    case StepsName(t)[Steps.BackingStorage]:
+      return validateBackingStorageStep(backingStorage, storageClass);
+    case StepsName(t)[Steps.CreateStorageClass]:
+      return canGoToNextStep && canGoToNextStep(createStorageClass, storageClass.name);
+    case StepsName(t)[Steps.ConnectionDetails]:
+      return canGoToNextStep && canGoToNextStep(createStorageClass, storageClass.name);
```

Comment:
https://github.com/openshift/console/pull/9502/files#diff-877dff9c6425dffffd9d09f7855a2e37bd8e9fc037347b4155d33eeb998f9c55R152

you are expecting only state as parameter. But here your are passing state as well as SC name. is this wont break?

Reply:
We are providing an API for users to consume. Its up to them, how they want to use.
Hence , its okay to consume the parameters that are required.
```
export const rhcsCanGoToNextStep: CanGoToNextStep<RHCSState> = (state) =>
  !!state.fileData && !state.isRejected && !state.isLoading;
  ```
It will not break anything, we are concerned with the output of `rhcsCanGoToNextStep`

Reply:
ack